tests/unit/auth.cpp

```diff
@@ -159,6 +160,73 @@ TEST_F(AuthWithStorage, UserRolePermissions) {
+  ASSERT_EQ(user->fine_grained_access_handler().label_permissions(), user->GetFineGrainedAccessLabelPermissions());
+  ASSERT_EQ(user->fine_grained_access_handler().edge_type_permissions(),
+            user->GetFineGrainedAccessEdgeTypePermissions());
+
+  // Grant one label to user .
+  user->fine_grained_access_handler().label_permissions().Grant("labelTest");
+  // Grant one edge type to user .
+  user->fine_grained_access_handler().edge_type_permissions().Grant("edgeTypeTest");
+
+  // Check permissions.
+  ASSERT_EQ(user->fine_grained_access_handler().label_permissions().Has("labelTest"), PermissionLevel::GRANT);
+  ASSERT_EQ(user->fine_grained_access_handler().edge_type_permissions().Has("edgeTypeTest"), PermissionLevel::GRANT);
+  ASSERT_EQ(user->fine_grained_access_handler().label_permissions(), user->GetFineGrainedAccessLabelPermissions());
+  ASSERT_EQ(user->fine_grained_access_handler().edge_type_permissions(),
+            user->GetFineGrainedAccessEdgeTypePermissions());
+
+  // Deny one label to user .
+  user->fine_grained_access_handler().label_permissions().Deny("labelTest1");
+  // Deny one edge type to user .
+  user->fine_grained_access_handler().edge_type_permissions().Deny("edgeTypeTest1");
+
+  // Check permissions.
+  ASSERT_EQ(user->fine_grained_access_handler().label_permissions().Has("labelTest1"), PermissionLevel::DENY);
+  ASSERT_EQ(user->fine_grained_access_handler().edge_type_permissions().Has("edgeTypeTest1"), PermissionLevel::DENY);
+  ASSERT_EQ(user->fine_grained_access_handler().label_permissions(), user->GetFineGrainedAccessLabelPermissions());
+  ASSERT_EQ(user->fine_grained_access_handler().edge_type_permissions(),
+            user->GetFineGrainedAccessEdgeTypePermissions());
+
+  // Create role.
+  ASSERT_TRUE(auth.AddRole("admin"));
+  auto role = auth.GetRole("admin");
+  ASSERT_NE(role, std::nullopt);
+
+  // Grant label and edge type to role and role to user.
+  role->fine_grained_access_handler().label_permissions().Grant("roleLabelTest");
+  role->fine_grained_access_handler().edge_type_permissions().Grant("roleEdgeTypeTest");
+  user->SetRole(*role);
+
+  // Check permissions.
+  {
+    ASSERT_EQ(user->GetFineGrainedAccessLabelPermissions().Has("roleLabelTest"), PermissionLevel::GRANT);
+    ASSERT_EQ(user->GetFineGrainedAccessEdgeTypePermissions().Has("roleEdgeTypeTest"), PermissionLevel::GRANT);
+  }
+
+  // Deny label and edge type to role and role to user.
+  role->fine_grained_access_handler().label_permissions().Deny("roleLabelTest1");
+  role->fine_grained_access_handler().edge_type_permissions().Deny("roleEdgeTypeTest1");
+  user->SetRole(*role);
+
+  // Check permissions.
+  {
+    ASSERT_EQ(user->GetFineGrainedAccessLabelPermissions().Has("roleLabelTest1"), PermissionLevel::DENY);
+    ASSERT_EQ(user->GetFineGrainedAccessEdgeTypePermissions().Has("roleEdgeTypeTest1"), PermissionLevel::DENY);
+  }
+}
+
```

Comment:
Do we need to check what if user has a grant and role has a deny? Is this covered? I'm not sure but I think it isn't, here in tests

Reply:
On the same label